src/librustc_passes/liveness.rs

```diff
@@ -1512,35 +1517,48 @@ impl<'tcx> Liveness<'_, 'tcx> {
             if is_assigned {
                 self.ir.tcx.lint_hir_note(
                     lint::builtin::UNUSED_VARIABLES,
-                    hir_id,
-                    spans,
+                    first_hir_id,
+                    hir_ids_and_spans.into_iter().map(|(_, sp)| sp).collect::<Vec<_>>(),
                     &format!("variable `{}` is assigned to, but never used", name),
                     &format!("consider using `_{}` instead", name),
                 );
             } else {
                 let mut err = self.ir.tcx.struct_span_lint_hir(
                     lint::builtin::UNUSED_VARIABLES,
-                    hir_id,
-                    spans.clone(),
+                    first_hir_id,
+                    hir_ids_and_spans.iter().map(|(_, sp)| *sp).collect::<Vec<_>>(),
                     &format!("unused variable: `{}`", name),
                 );
 
                 if self.ir.variable_is_shorthand(var) {
```

Comment:
This should be checked for each variable in the pattern: `A{ x } | B { f: x }` is allowed.

Reply:
Which warning should be printed in that case?

Reply:
"try ignoring these fields..." "...and prefixing these with an underscore"?